browser/brave_wallet/brave_wallet_event_emitter_browsertest.cc

```diff
@@ -142,3 +182,33 @@ IN_PROC_BROWSER_TEST_F(BraveWalletEventEmitterTest,
              content::EXECUTE_SCRIPT_USE_MANUAL_REPLY);
   EXPECT_EQ(base::Value(true), result_first.value);
 }
+
+IN_PROC_BROWSER_TEST_F(BraveWalletEventEmitterTest,
+                       CheckForAnAccountChangedEvent) {
+  RestoreWallet();
+  GURL url =
+      https_server()->GetURL("a.com", "/brave_wallet_event_emitter.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  std::string address;
+  GetAddress(&address);
+
+  url::Origin sub_request_origin;
+  ASSERT_TRUE(brave_wallet::GetSubRequestOrigin(
+      permissions::RequestType::kBraveEthereum, GetLastCommitedOrigin(),
+      address, &sub_request_origin));
+  host_content_settings_map()->SetContentSettingDefaultScope(
+      sub_request_origin.GetURL(), GetLastCommitedOrigin().GetURL(),
+      ContentSettingsType::BRAVE_ETHEREUM,
+      ContentSetting::CONTENT_SETTING_ALLOW);
```

Comment:
We don't need to modify content setting to get account changed event triggered. `SetSelectedAccount` solely would do the job.

Reply:
SetSelected account doesn't set an account that doesn't have a permission

Reply:
That is not true,
```
void KeyringService::SetSelectedAccountForCoin(mojom::CoinType coin,
                                               const std::string& address) {
  SetPrefForKeyring(prefs_, kSelectedAccount, base::Value(address),
                    GetKeyringIdForCoin(coin));
  NotifySelectedAccountChanged(coin);
}
```
where is the permission check?

Reply:
```
void EthereumProviderImpl::OnUpdateKnownAccounts(
    const std::vector<std::string>& allowed_accounts,
    mojom::ProviderError error,
    const std::string& error_message) {
  if (error != mojom::ProviderError::kSuccess) {
    return;
  }
  bool accounts_changed = allowed_accounts != known_allowed_accounts;
```
`known_allowed_accounts` are empty after `SetSelectedAccount`, I need them there and they are set only after we allow some account

Reply:
I see. you are right that we need permission first to get the event fired